fix: v2sku for APIM - `avm/res/api-management/service`

tony-box opened the pull request:
## Description

<!--
>Thank you for your contribution !
> Please include a summary of the change and which issue is fixed.
> Please also include the context.
> List any dependencies that are required for this change.

Fixes #123
Fixes #456
Closes #123
Closes #456
-->

Closes #1929 
Closes #2075 
Closes #1473
- Update all api-management/service/* modules to use newer API versions (2022-08-01)
- Update api-management/service api version 2023-05-01-preview to allow v2sku options
- Add V2sku deployment option for api-management/service
- adds an additional test for v2 sku deployment.
- modify all tests to deploy APIM with Premium SKU, and with multiple zones and 2nd region to appease latest PS Rule tests.

## Pipeline Reference

<!-- Insert your Pipeline Status Badge below -->

| Pipeline |
| -------- |
|    [![avm.res.api-management.service](https://github.com/tony-box/bicep-registry-modules/actions/workflows/avm.res.api-management.service.yml/badge.svg?branch=fix%2Fv2sku)](https://github.com/tony-box/bicep-registry-modules/actions/workflows/avm.res.api-management.service.yml)      |

## Type of Change

<!-- Use the check-boxes [x] on the options that are relevant. -->

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [x] Azure Verified Module updates:
  - [x] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [x] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [x] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

<!--  Please keep up to day with the contribution guide at https://aka.ms/avm/contribute/bicep -->

eriqua commented on .github/ISSUE_TEMPLATE/avm_module_issue.yml:
```diff
@@ -45,7 +45,6 @@ body:
         # - "avm/ptn/avd-lza/management-plane"
         # - "avm/ptn/avd-lza/networking"
         # - "avm/ptn/avd-lza/session-hosts"
-        - "avm/ptn/policy-insights/remediation"
```
Is this removal intended?

tony-box replied:
@eriqua I fixed this. my fork was messed up and had some weird commits in it that were getting dragged along into all my other branches. This should be a clean commit now with just the changes related to the fix.

eriqua commented:
@tony-box please also have a look at #1795 which is also targeting the same resource module and may thus result in conflicts to be addressed. 
I'd suggest to prioritize addressing comments there first

tony-box commented:
> @tony-box please also have a look at #1795 which is also targeting the same resource module and may thus result in conflicts to be addressed. I'd suggest to prioritize addressing comments there first

There shouldn't be any conflicts between these two PRs. Regardless, I pretty much have to redo that whole PR anyway due to all the weird commits that were included in it (similar to this one). I will be retesting it and ensuring there are no issues before that one goes through.

ChrisSidebotham commented:
Converting back to draft until @tony-box is ready :)

tony-box commented:
I'm going to update the rest of the APIM services' API version in this PR to fix #2075 as well.

AlexanderSehr commented on avm/res/api-management/service/tests/e2e/defaults/main.test.bicep:
```diff
@@ -45,6 +45,19 @@ module testDeployment '../../../main.bicep' = [
       location: resourceLocation
       publisherEmail: 'apimgmt-noreply@mail.windowsazure.com'
       publisherName: '${namePrefix}-az-amorg-x-001'
+      sku: 'Premium'
+      skuCount: 2
+      zones: ['1', '2']
+      additionalLocations: [
+        {
+          location: 'westus'
+          sku: {
+            name: 'Premium'
+            capacity: 1
+          }
+          disableGateway: false
+        }
+      ]
```
Are those parameters required?

AlexanderSehr commented:
Hey @tony-box, I see there was some back and forth regarding this PR. Would you say it's ready?

tony-box commented:
closing this PR which will be superceeded by #2453
